File: Contests/game11.cpp
#include "bits/stdc++.h"
using namespace std;
int main() {
  int t;
  cin >> t;
  while (t--) {
    int n, m;
    cin >> n >> m;
    vector<int> a(n), b(m);
    for (int i = 0; i < n; i++) {
      cin >> a[i];
    }
    for (int i = 0; i < m; i++) {
      cin >> b[i];
    }
    if (a.size() < 4 or b.size() < 4) {
      cout << "-1\n";
    }
    sort(a.begin(), a.end()), sort(b.begin(), b.end());
    int cnt = 0;
    for (int i = n - 1; i >= 4; i--) {
      cnt++;
      erase(a.begin(), a.end(), a[i]);
    }
  }
}

File: Contests/game11.java
import java.util.*;
import java.lang.*;
import java.io.*;

class Codechef {
  public static void main(String[] args) throws java.lang.Exception {
    Scanner scanner = new Scanner(System.in);
    int T = scanner.nextInt();
    for (int i = 0; i < T; i++) {
      int N = scanner.nextInt();
      int M = scanner.nextInt();
      long[] batsmenSkills = new long[N];
      long[] bowlersSkills = new long[M];
      for (int j = 0; j < N; j++) {
        batsmenSkills[j] = scanner.nextLong();
      }
      for (int j = 0; j < M; j++) {
        bowlersSkills[j] = scanner.nextLong();
      }
      Arrays.sort(batsmenSkills);
      Arrays.sort(bowlersSkills);
      if (N < 4 || M < 4 || (M + N) < 11) {
        System.out.println("-1");
        continue;
      }
      long maxSkill = 0;
      int j = N - 1;
      int k = M - 1;
      for (int cnt = 0; cnt < 4; cnt++) {
        maxSkill += batsmenSkills[j--];
      }
      for (int cnt = 0; cnt < 4; cnt++) {
        maxSkill += bowlersSkills[k--];
      }
      for (int cnt = 0; cnt < 3; cnt++) {
        if (j >= 0 && k >= 0) {
          if (batsmenSkills[j] >= bowlersSkills[k]) {
            maxSkill += batsmenSkills[j--];
          } else {
            maxSkill += bowlersSkills[k--];
          }
        } else if (j >= 0) {
          maxSkill += batsmenSkills[j--];
        } else {
          maxSkill += bowlersSkills[k--];
        }
      }
      System.out.println(maxSkill);
    }
    scanner.close();
  }
}
